### PR TITLE
Add support for Python 3.6 (SCL)

### PR DESCRIPTION
**Pull request by joshmoore**
cc: @manics

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -3,8 +3,17 @@
...
-cp dist/* /dist/
+
+build() {
+    mkdir /tmp/$1
+    cd /tmp/$1
+    pip download "zeroc-ice==$ICE_VERSION"
+    tar -zxf "zeroc-ice-$ICE_VERSION.tar.gz"
+    cd "zeroc-ice-$ICE_VERSION"
+    python setup.py bdist_wheel bdist_rpm
+    cp dist/* /dist/
```
Comment (manics) [inline build.sh:14]:
```
moving build/bdist.linux-x86_64/rpm/SRPMS/zeroc-ice-3.6.4-1.src.rpm -> dist
moving build/bdist.linux-x86_64/rpm/RPMS/x86_64/zeroc-ice-3.6.4-1.x86_64.rpm -> dist
moving build/bdist.linux-x86_64/rpm/RPMS/x86_64/zeroc-ice-debuginfo-3.6.4-1.x86_64.rpm -> dist
```
The artifacts have the same name for both builds so they overwrite each other.

Comment (manics) [reply build.sh:14]: Though I think the wheels have different names so could disable `bdist_rpm`?

**Comment by joshmoore** [comment]
Currently attempting wheel only and only for py3.6.  If that works, propose to release as 0.2.0

**Comment by manics** [comment]
FYI `python27-python-wheel` and `rh-python36-python-wheel` packages exist in the yum repos, no idea if there was a reason for `pip install wheel` of it it simply wasn't available at the time.

**Comment by joshmoore** [comment]
<details>
<summary>Importing Ice in the same docker image that built it leads to a segfault</summary>
<div>
<pre>
root@4abe45f20120 /]# python
Python 3.6.3 (default, May 31 2019, 13:05:43)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import Ice
>>> Ice.stringVersion()
'3.6.4'
>>>
*** Error in `python': free(): invalid pointer: 0x00007f117d861258 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81609)[0x7f11816f7609]
/opt/rh/rh-python36/root/lib64/python3.6/site-packages/IcePy.cpython-36m-x86_64-linux-gnu.so(_ZN5IcePy14PyObjectHandleaSEP7_object+0x2b)[0x7f11800c1c7b]
/opt/rh/rh-python36/root/lib64/python3.6/site-packages/IcePy.cpython-36m-x86_64-linux-gnu.so(_ZN5IcePy16InfoMapDestroyerD1Ev+0x33)[0x7f11800dd6d3]
/lib64/libc.so.6(+0x39c29)[0x7f11816afc29]
/lib64/libc.so.6(+0x39c77)[0x7f11816afc77]
/lib64/libc.so.6(__libc_start_main+0xfc)[0x7f118169849c]
python[0x400b90]
======= Memory map: ========
00400000-00401000 r-xp 00000000 fd:05 53485360                           /opt/rh/rh-python36/root/usr/bin/python3.6
00601000-00602000 r--p 00001000 fd:05 53485360                           /opt/rh/rh-python36/root/usr/bin/python3.6
00602000-00603000 rw-p 00002000 fd:05 53485360                           /opt/rh/rh-python36/root/usr/bin/python3.6
01d23000-01eb3000 rw-p 00000000 00:00 0                                  [heap]
7f1178000000-7f1178021000 rw-p 00000000 00:00 0
7f1178021000-7f117c000000 ---p 00000000 00:00 0
7f117d7a6000-7f117d866000 rw-p 00000000 00:00 0
7f117d866000-7f117d8c6000 r-xp 00000000 fd:05 59664062                   /usr/lib64/libpcre.so.1.2.0
7f117d8c6000-7f117dac6000 ---p 00060000 fd:05 59664062                   /usr/lib64/libpcre.so.1.2.0
7f117dac6000-7f117dac7000 r--p 00060000 fd:05 59664062                   /usr/lib64/libpcre.so.1.2.0
7f117dac7000-7f117dac8000 rw-p 00061000 fd:05 59664062                   /usr/lib64/libpcre.so.1.2.0
7f117dac8000-7f117daec000 r-xp 00000000 fd:05 53494236                   /usr/lib64/libselinux.so.1
7f117daec000-7f117dceb000 ---p 00024000 fd:05 53494236                   /usr/lib64/libselinux.so.1
7f117dceb000-7f117dcec000 r--p 00023000 fd:05 53494236                   /usr/lib64/libselinux.so.1
7f117dcec000-7f117dced000 rw-p 00024000 fd:05 53494236                   /usr/lib64/libselinux.so.1
7f117dced000-7f117dcef000 rw-p 00000000 00:00 0
7f117dcef000-7f117dd05000 r-xp 00000000 fd:05 53494224                   /usr/lib64/libresolv-2.17.so
7f117dd05000-7f117df04000 ---p 00016000 fd:05 53494224                   /usr/lib64/libresolv-2.17.so
7f117df04000-7f117df05000 r--p 00015000 fd:05 53494224                   /usr/lib64/libresolv-2.17.so
7f117df05000-7f117df06000 rw-p 00016000 fd:05 53494224                   /usr/lib64/libresolv-2.17.so
7f117df06000-7f117df08000 rw-p 00000000 00:00 0
7f117df08000-7f117df0b000 r-xp 00000000 fd:05 59663982                   /usr/lib64/libkeyutils.so.1.5
7f117df0b000-7f117e10a000 ---p 00003000 fd:05 59663982                   /usr/lib64/libkeyutils.so.1.5
7f117e10a000-7f117e10b000 r--p 00002000 fd:05 59663982                   /usr/lib64/libkeyutils.so.1.5
7f117e10b000-7f117e10c000 rw-p 00003000 fd:05 59663982                   /usr/lib64/libkeyutils.so.1.5
7f117e10c000-7f117e11a000 r-xp 00000000 fd:05 53494171                   /usr/lib64/libkrb5support.so.0.1
7f117e11a000-7f117e31a000 ---p 0000e000 fd:05 53494171                   /usr/lib64/libkrb5support.so.0.1
7f117e31a000-7f117e31b000 r--p 0000e000 fd:05 53494171                   /usr/lib64/libkrb5support.so.0.1
7f117e31b000-7f117e31c000 rw-p 0000f000 fd:05 53494171                   /usr/lib64/libkrb5support.so.0.1
7f117e31c000-7f117e331000 r-xp 00000000 fd:05 53494317                   /usr/lib64/libz.so.1.2.7
7f117e331000-7f117e530000 ---p 00015000 fd:05 53494317                   /usr/lib64/libz.so.1.2.7
7f117e530000-7f117e531000 r--p 00014000 fd:05 53494317                   /usr/lib64/libz.so.1.2.7
7f117e531000-7f117e532000 rw-p 00015000 fd:05 53494317                   /usr/lib64/libz.so.1.2.7
7f117e532000-7f117e563000 r-xp 00000000 fd:05 53494152                   /usr/lib64/libk5crypto.so.3.1
7f117e563000-7f117e762000 ---p 00031000 fd:05 53494152                   /usr/lib64/libk5crypto.so.3.1
7f117e762000-7f117e764000 r--p 00030000 fd:05 53494152                   /usr/lib64/libk5crypto.so.3.1
7f117e764000-7f117e765000 rw-p 00032000 fd:05 53494152                   /usr/lib64/libk5crypto.so.3.1
7f117e765000-7f117e768000 r-xp 00000000 fd:05 53494071                   /usr/lib64/libcom_err.so.2.1
7f117e768000-7f117e967000 ---p 00003000 fd:05 53494071                   /usr/lib64/libcom_err.so.2.1
7f117e967000-7f117e968000 r--p 00002000 fd:05 53494071                   /usr/lib64/libcom_err.so.2.1
7f117e968000-7f117e969000 rw-p 00003000 fd:05 53494071                   /usr/lib64/libcom_err.so.2.1
7f117e969000-7f117ea42000 r-xp 00000000 fd:05 53494169                   /usr/lib64/libkrb5.so.3.3
7f117ea42000-7f117ec41000 ---p 000d9000 fd:05 53494169                   /usr/lib64/libkrb5.so.3.3
7f117ec41000-7f117ec4f000 r--p 000d8000 fd:05 53494169                   /usr/lib64/libkrb5.so.3.3
7f117ec4f000-7f117ec52000 rw-p 000e6000 fd:05 53494169                   /usr/lib64/libkrb5.so.3.3
7f117ec52000-7f117ec9c000 r-xp 00000000 fd:05 53494136                   /usr/lib64/libgssapi_krb5.so.2.2
7f117ec9c000-7f117ee9c000 ---p 0004a000 fd:05 53494136                   /usr/lib64/libgssapi_krb5.so.2.2
7f117ee9c000-7f117ee9d000 r--p 0004a000 fd:05 53494136                   /usr/lib64/libgssapi_krb5.so.2.2
7f117ee9d000-7f117ee9f000 rw-p 0004b000 fd:05 53494136                   /usr/lib64/libgssapi_krb5.so.2.2
7f117ee9f000-7f117eeb4000 r-xp 00000000 fd:05 53494116                   /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f117eeb4000-7f117f0b3000 ---p 00015000 fd:05 53494116                   /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f117f0b3000-7f117f0b4000 r--p 00014000 fd:05 53494116                   /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f117f0b4000-7f117f0b5000 rw-p 00015000 fd:05 53494116                   /usr/lib64/libgcc_s-4.8.5-20150702.so.1
7f117f0b5000-7f117f19e000 r-xp 00000000 fd:05 53494241                   /usr/lib64/libstdc++.so.6.0.19
7f117f19e000-7f117f39d000 ---p 000e9000 fd:05 53494241                   /usr/lib64/libstdc++.so.6.0.19
7f117f39d000-7f117f3a5000 r--p 000e8000 fd:05 53494241                   /usr/lib64/libstdc++.so.6.0.19
7f117f3a5000-7f117f3a7000 rw-p 000f0000 fd:05 53494241                   /usr/lib64/libstdc++.so.6.0.19
7f117f3a7000-7f117f3bc000 rw-p 00000000 00:00 0
7f117f3bc000-7f117f3c3000 r-xp 00000000 fd:05 53494231                   /usr/lib64/librt-2.17.so
7f117f3c3000-7f117f5c2000 ---p 00007000 fd:05 53494231                   /usr/lib64/librt-2.17.so
7f117f5c2000-7f117f5c3000 r--p 00006000 fd:05 53494231                   /usr/lib64/librt-2.17.so
7f117f5c3000-7f117f5c4000 rw-p 00007000 fd:05 53494231                   /usr/lib64/librt-2.17.so
7f117f5c4000-7f117f5d3000 r-xp 00000000 fd:05 59663893                   /usr/lib64/libbz2.so.1.0.6
7f117f5d3000-7f117f7d2000 ---p 0000f000 fd:05 59663893                   /usr/lib64/libbz2.so.1.0.6
7f117f7d2000-7f117f7d3000 r--p 0000e000 fd:05 59663893                   /usr/lib64/libbz2.so.1.0.6
7f117f7d3000-7f117f7d4000 rw-p 0000f000 fd:05 59663893                   /usr/lib64/libbz2.so.1.0.6
7f117f7d4000-7f117fa09000 r-xp 00000000 fd:05 53494075                   /usr/lib64/libcrypto.so.1.0.2k
7f117fa09000-7f117fc09000 ---p 00235000 fd:05 53494075                   /usr/lib64/libcrypto.so.1.0.2k
7f117fc09000-7f117fc25000 r--p 00235000 fd:05 53494075                   /usr/lib64/libcrypto.so.1.0.2k
7f117fc25000-7f117fc32000 rw-p 00251000 fd:05 53494075                   /usr/lib64/libcrypto.so.1.0.2k
7f117fc32000-7f117fc36000 rw-p 00000000 00:00 0
7f117fc36000-7f117fc9d000 r-xp 00000000 fd:05 53494240                   /usr/lib64/libssl.so.1.0.2k
7f117fc9d000-7f117fe9d000 ---p 00067000 fd:05 53494240                   /usr/lib64/libssl.so.1.0.2k
7f117fe9d000-7f117fea1000 r--p 00067000 fd:05 53494240                   /usr/lib64/libssl.so.1.0.2k
7f117fea1000-7f117fea8000 rw-p 0006b000 fd:05 53494240                   /usr/lib64/libssl.so.1.0.2k
7f117fea8000-7f118056f000 r-xp 00000000 fd:05 54920500                   /opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/IcePy.cpython-36m-x86_64-linux-gnu.so
7f118056f000-7f118076e000 ---p 006c7000 fd:05 54920500                   /opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/IcePy.cpython-36m-x86_64-linux-gnu.so
7f118076e000-7f11807c9000 r--p 006c6000 fd:05 54920500                   /opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/IcePy.cpython-36m-x86_64-linux-gnu.so
7f11807c9000-7f11807da000 rw-p 00721000 fd:05 54920500                   /opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/IcePy.cpython-36m-x86_64-linux-gnu.so
7f11807da000-7f118084a000 rw-p 00000000 00:00 0
7f118084a000-7f1180860000 r-xp 00000000 fd:05 53488754                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_datetime.cpython-36m-x86_64-linux-gnu.so
7f1180860000-7f1180a60000 ---p 00016000 fd:05 53488754                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_datetime.cpython-36m-x86_64-linux-gnu.so
7f1180a60000-7f1180a61000 r--p 00016000 fd:05 53488754                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_datetime.cpython-36m-x86_64-linux-gnu.so
7f1180a61000-7f1180a64000 rw-p 00017000 fd:05 53488754                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_datetime.cpython-36m-x86_64-linux-gnu.so
7f1180a64000-7f1180a6e000 r-xp 00000000 fd:05 53488782                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/math.cpython-36m-x86_64-linux-gnu.so
7f1180a6e000-7f1180c6d000 ---p 0000a000 fd:05 53488782                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/math.cpython-36m-x86_64-linux-gnu.so
7f1180c6d000-7f1180c6e000 r--p 00009000 fd:05 53488782                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/math.cpython-36m-x86_64-linux-gnu.so
7f1180c6e000-7f1180c70000 rw-p 0000a000 fd:05 53488782                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/math.cpython-36m-x86_64-linux-gnu.so
7f1180c70000-7f1180d79000 rw-p 00000000 00:00 0
7f1180d79000-7f1180d7c000 r-xp 00000000 fd:05 53488760                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_heapq.cpython-36m-x86_64-linux-gnu.so
7f1180d7c000-7f1180f7b000 ---p 00003000 fd:05 53488760                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_heapq.cpython-36m-x86_64-linux-gnu.so
7f1180f7b000-7f1180f7c000 r--p 00002000 fd:05 53488760                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_heapq.cpython-36m-x86_64-linux-gnu.so
7f1180f7c000-7f1180f7e000 rw-p 00003000 fd:05 53488760                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/_heapq.cpython-36m-x86_64-linux-gnu.so
7f1180f7e000-7f1180ffe000 rw-p 00000000 00:00 0
7f1180ffe000-7f1181023000 r-xp 00000000 fd:05 59664136                   /usr/lib64/libtinfo.so.5.9
7f1181023000-7f1181223000 ---p 00025000 fd:05 59664136                   /usr/lib64/libtinfo.so.5.9
7f1181223000-7f1181227000 r--p 00025000 fd:05 59664136                   /usr/lib64/libtinfo.so.5.9
7f1181227000-7f1181228000 rw-p 00029000 fd:05 59664136                   /usr/lib64/libtinfo.so.5.9
7f1181228000-7f1181264000 r-xp 00000000 fd:05 59664087                   /usr/lib64/libreadline.so.6.2
7f1181264000-7f1181464000 ---p 0003c000 fd:05 59664087                   /usr/lib64/libreadline.so.6.2
7f1181464000-7f1181466000 r--p 0003c000 fd:05 59664087                   /usr/lib64/libreadline.so.6.2
7f1181466000-7f118146c000 rw-p 0003e000 fd:05 59664087                   /usr/lib64/libreadline.so.6.2
7f118146c000-7f118146e000 rw-p 00000000 00:00 0
7f118146e000-7f1181474000 r-xp 00000000 fd:05 53488788                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/readline.cpython-36m-x86_64-linux-gnu.so
7f1181474000-7f1181673000 ---p 00006000 fd:05 53488788                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/readline.cpython-36m-x86_64-linux-gnu.so
7f1181673000-7f1181674000 r--p 00005000 fd:05 53488788                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/readline.cpython-36m-x86_64-linux-gnu.so
7f1181674000-7f1181676000 rw-p 00006000 fd:05 53488788                   /opt/rh/rh-python36/root/usr/lib64/python3.6/lib-dynload/readline.cpython-36m-x86_64-linux-gnu.so
7f1181676000-7f1181838000 r-xp 00000000 fd:05 53494061                   /usr/lib64/libc-2.17.so
7f1181838000-7f1181a38000 ---p 001c2000 fd:05 53494061                   /usr/lib64/libc-2.17.so
7f1181a38000-7f1181a3c000 r--p 001c2000 fd:05 53494061                   /usr/lib64/libc-2.17.so
7f1181a3c000-7f1181a3e000 rw-p 001c6000 fd:05 53494061                   /usr/lib64/libc-2.17.so
7f1181a3e000-7f1181a43000 rw-p 00000000 00:00 0
7f1181a43000-7f1181b44000 r-xp 00000000 fd:05 53494172                   /usr/lib64/libm-2.17.so
7f1181b44000-7f1181d43000 ---p 00101000 fd:05 53494172                   /usr/lib64/libm-2.17.so
7f1181d43000-7f1181d44000 r--p 00100000 fd:05 53494172                   /usr/lib64/libm-2.17.so
7f1181d44000-7f1181d45000 rw-p 00101000 fd:05 53494172                   /usr/lib64/libm-2.17.so
7f1181d45000-7f1181d47000 r-xp 00000000 fd:05 53494252                   /usr/lib64/libutil-2.17.so
7f1181d47000-7f1181f46000 ---p 00002000 fd:05 53494252                   /usr/lib64/libutil-2.17.so
7f1181f46000-7f1181f47000 r--p 00001000 fd:05 53494252                   /usr/lib64/libutil-2.17.so
7f1181f47000-7f1181f48000 rw-p 00002000 fd:05 53494252                   /usr/lib64/libutil-2.17.so
7f1181f48000-7f1181f4a000 r-xp 00000000 fd:05 53494086                   /usr/lib64/libdl-2.17.so
7f1181f4a000-7f118214a000 ---p 00002000 fd:05 53494086                   /usr/lib64/libdl-2.17.so
7f118214a000-7f118214b000 r--p 00002000 fd:05 53494086                   /usr/lib64/libdl-2.17.so
7f118214b000-7f118214c000 rw-p 00003000 fd:05 53494086                   /usr/lib64/libdl-2.17.so
7f118214c000-7f1182163000 r-xp 00000000 fd:05 53494219                   /usr/lib64/libpthread-2.17.so
7f1182163000-7f1182362000 ---p 00017000 fd:05 53494219                   /usr/lib64/libpthread-2.17.so
7f1182362000-7f1182363000 r--p 00016000 fd:05 53494219                   /usr/lib64/libpthread-2.17.so
7f1182363000-7f1182364000 rw-p 00017000 fd:05 53494219                   /usr/lib64/libpthread-2.17.so
7f1182364000-7f1182368000 rw-p 00000000 00:00 0
7f1182368000-7f11825f5000 r-xp 00000000 fd:05 53486535                   /opt/rh/rh-python36/root/usr/lib64/libpython3.6m.so.rh-python36-1.0
7f11825f5000-7f11827f4000 ---p 0028d000 fd:05 53486535                   /opt/rh/rh-python36/root/usr/lib64/libpython3.6m.so.rh-python36-1.0
7f11827f4000-7f11827f7000 r--p 0028c000 fd:05 53486535                   /opt/rh/rh-python36/root/usr/lib64/libpython3.6m.so.rh-python36-1.0
7f11827f7000-7f118285d000 rw-p 0028f000 fd:05 53486535                   /opt/rh/rh-python36/root/usr/lib64/libpython3.6m.so.rh-python36-1.0
7f118285d000-7f118288f000 rw-p 00000000 00:00 0
7f118288f000-7f11828b1000 r-xp 00000000 fd:05 53493930                   /usr/lib64/ld-2.17.so
7f11828be000-7f11828fe000 rw-p 00000000 00:00 0
7f1182923000-7f1182aa8000 rw-p 00000000 00:00 0
7f1182aad000-7f1182ab0000 rw-p 00000000 00:00 0
7f1182ab0000-7f1182ab1000 r--p 00021000 fd:05 53493930                   /usr/lib64/ld-2.17.so
7f1182ab1000-7f1182ab2000 rw-p 00022000 fd:05 53493930                   /usr/lib64/ld-2.17.so
7f1182ab2000-7f1182ab3000 rw-p 00000000 00:00 0
7ffcae83f000-7ffcae860000 rw-p 00000000 00:00 0                          [stack]
7ffcae9ef000-7ffcae9f1000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted (core dumped)
</pre>
</div>
</details>

**Comment by manics** [comment]
Could try activating the python3 environment instead of just setting the path (for both the build and the test)?

**Comment by sbesson** [comment]
Closing as this is superseded by #4

**Pull request closed without merging**